src/parsing/staticallyCompileXPath.ts

```diff
@@ -44,69 +52,153 @@ export default function staticallyCompileXPath(
+	moduleImports: { [namespaceURI: string]: string },
+	defaultFunctionNamespaceURI: string,
+	functionNameResolver: FunctionNameResolver,
+	rootStaticContext: StaticContext
+) {
+	const context = new AnnotationContext(rootStaticContext);
+	if (compilationOptions.annotateAst) {
+		annotateAst(ast, context);
+	}
 
-		if (prolog) {
-			if (!compilationOptions.allowXQuery) {
-				throw new Error(
-					'XPST0003: Use of XQuery functionality is not allowed in XPath context'
-				);
-			}
-			processProlog(prolog, rootStaticContext);
-		}
+	const mainModule = astHelper.getFirstChild(ast, 'mainModule');
+	if (!mainModule) {
+		// This must be a library module
+		throw new Error('Can not execute a library module.');
+	}
+
+	const prolog = astHelper.getFirstChild(mainModule, 'prolog');
+	const queryBodyContents = astHelper.followPath(mainModule, ['queryBody', '*']);
 
-		expression = compileAstToExpression(queryBodyContents, compilationOptions);
+	if (prolog) {
+		if (!compilationOptions.allowXQuery) {
+			throw new Error(
+				'XPST0003: Use of XQuery functionality is not allowed in XPath context'
+			);
+		}
+		processProlog(prolog, rootStaticContext);
 	}
 
-	if (fromCache === null || fromCache.requiresStaticCompilation) {
-		Object.keys(moduleImports).forEach((modulePrefix) => {
-			const moduleURI = moduleImports[modulePrefix];
-			enhanceStaticContextWithModule(rootStaticContext, moduleURI);
+	return compileAstToExpression(queryBodyContents, compilationOptions);
+}
 
-			rootStaticContext.registerNamespace(modulePrefix, moduleURI);
-		});
+export default function staticallyCompileXPath(
+	selector: EvaluableExpression,
+	compilationOptions: {
+		allowUpdating: boolean | undefined;
+		allowXQuery: boolean | undefined;
+		annotateAst: boolean | undefined;
+		debug: boolean | undefined;
+		disableCache: boolean | undefined;
+	},
+	namespaceResolver: (namespace: string) => string | null,
+	variables: { [varName: string]: any },
+	moduleImports: { [namespaceURI: string]: string },
+	defaultFunctionNamespaceURI: string,
+	functionNameResolver: FunctionNameResolver
+): { expression: Expression; staticContext: StaticContext } {
+	const executionSpecificStaticContext = new ExecutionSpecificStaticContext(
+		namespaceResolver,
+		variables,
+		defaultFunctionNamespaceURI,
+		functionNameResolver
+	);
+	const rootStaticContext = new StaticContext(executionSpecificStaticContext);
+	Object.keys(moduleImports).forEach((modulePrefix) => {
+		const moduleURI = moduleImports[modulePrefix];
+		enhanceStaticContextWithModule(rootStaticContext, moduleURI);
 
-		expression.performStaticEvaluation(rootStaticContext);
+		rootStaticContext.registerNamespace(modulePrefix, moduleURI);
+	});
 
-		if (!compilationOptions.disableCache) {
-			storeStaticCompilationResultInCache(
-				xpathString,
-				language,
-				executionSpecificStaticContext,
-				moduleImports,
-				expression,
-				compilationOptions.debug,
-				defaultFunctionNamespaceURI
-			);
```

Comment:
I don't see where this moved to - are we not storing this in the cache anymore?

Reply:
That is correct. We need to move this code to line 179 in the new code. This also explains the big drop in coverage!